components/sup/src/manager/mod.rs

```diff
@@ -145,6 +146,13 @@ impl Manager {
         loop {
             let next_check = SteadyTime::now() + TimeDuration::milliseconds(1000);
             if self.check_for_incoming_signals() {
+                let mut services = self.state
+                    .services
+                    .write()
+                    .expect("Services lock is poisend!");
+                for service in services.drain(..) {
```

Comment:
I love this. `drain` creates an iterator for you *and* clears the vec at the same time